unstructured_inference/models/chipper.py

```diff
@@ -1,173 +1,502 @@
+
+        self.input_ids = self.processor.tokenizer(
+            prompt,
+            add_special_tokens=False,
+            return_tensors="pt",
+        ).input_ids.to(self.device)
+
+        self.start_tokens = [
+            v
+            for k, v in self.processor.tokenizer.get_added_vocab().items()
+            if k.startswith("<s_") and v not in self.input_ids
+        ]
+        self.end_tokens = [
+            v for k, v in self.processor.tokenizer.get_added_vocab().items() if k.startswith("</s_")
+        ]
+        self.tokens_stop = [self.tokenizer.eos_token_id, self.tokenizer.pad_token_id]
 
-    def predict(self, image) -> List[LocationlessLayoutElement]:
+        self.model.to(self.device)
+        self.model.eval()
+
+    def predict(self, image) -> List[LayoutElement]:
         """Do inference using the wrapped model."""
-        tokens = self.predict_tokens(image)
-        elements = self.postprocess(tokens)
+        tokens, decoder_cross_attentions = self.predict_tokens(image)
+        elements = self.postprocess(image, tokens, decoder_cross_attentions)
         return elements
 
-    def predict_tokens(self, image: Image) -> List[int]:
+    def predict_tokens(
+        self,
+        image: Image,
+    ) -> Tuple[List[int], Sequence[Sequence[torch.Tensor]]]:
         """Predict tokens from image."""
-        annotation = self.model.generate(
-            self.processor(
-                np.array(
-                    image,
-                    np.float32,
-                ),
-                return_tensors="pt",
-            ).pixel_values,
-            decoder_input_ids=torch.tensor([[0]]),
-            do_sample=True,
-            top_p=0.92,
-            top_k=0,
-            no_repeat_ngram_size=10,
-            num_beams=3,
-        ).tolist()[0]
-
-        tokens = (
-            [self.processor.tokenizer.bos_token_id]
-            + [
-                e
-                for e in annotation
-                if e != self.processor.tokenizer.bos_token_id
-                and e != self.processor.tokenizer.eos_token_id
-            ]
-            + [self.processor.tokenizer.eos_token_id]
-        )
+        with torch.no_grad():
+            outputs = self.model.generate(
+                self.processor(
+                    np.array(
+                        image,
+                        np.float32,
+                    ),
+                    return_tensors="pt",
+                ).pixel_values.to(self.device),
+                decoder_input_ids=self.input_ids,
+                logits_processor=[self.logits_processor],
+                # max_length=self.max_length,
+                do_sample=True,
+                top_p=0.92,
+                top_k=5,
+                no_repeat_ngram_size=0,
+                num_beams=3,
+                return_dict_in_generate=True,
+                output_attentions=True,
+                output_scores=True,
+                output_hidden_states=False,
```

Comment:
remove the commented line? and maybe make a constant for this default config?

Reply:
The max_length should be uncommented. It has been corrected.